Neos.Flow/Classes/Core/Booting/Scripts.php

```diff
@@ -736,9 +741,33 @@ public static function buildPhpCommand(array $settings)
             $command .= ' -c ' . escapeshellarg($useIniFile);
         }
 
+
         return $command;
     }
 
+    /**
+     * @param string phpBinaryPathAndFilename
+     * @throws Exception
+     */
+    protected static function ensureCurrentPhpVersionMatchesConfiguredVersion($phpBinaryPathAndFilename)
+    {
+        // Resolve any symlinks that the configured php might be pointing to
+        $configuredPhpBinaryPathAndFilename = realpath($phpBinaryPathAndFilename);
+
+        // if the configured PHP binary is empty here, the file does not exist. We ignore that here because it is checked later in the script.
+        if (strlen($configuredPhpBinaryPathAndFilename) === 0) {
+            return;
+        }
+
+        if (strcmp(PHP_BINARY, $configuredPhpBinaryPathAndFilename) !== 0) {
```

Comment:
I'll have to check if this works on windows.

Reply:
So, yeah - this is what I thought would happen and probably also the entire reason we have that config setting.
![image](https://user-images.githubusercontent.com/10347669/45280760-f23e4100-b4d5-11e8-9603-66c59b2168c3.png)

I see two ways forward here:
1. Do the check only for CLI requests. This would make the error message disappear, but again open the possibility for hard-to-detect configuration errors
2. Do the check as-is for CLI requests, and do the check like before for non-CLI (run exec(), checking the PHP version of the configured PHP-CLI, and compare the actual version number with the running one). 

I prefer the second option, as it makes config errors impossible, gives the user a precise and actionable error message, and has a minimal performance impact (delays the building of proxies by one exec(), but since proxy building takes a few secs at minimum anyway, this should be hardly noticeable).

Reply:
I'm not so much concerned about performance. but option 2 spawns an additional process for every sub request. For packages that use that mechanism extensively (JobQueue, EventSourcing, ...) that doubles the amount of processes. I don't know how problematic that is, but we should evaluate that IMO.

Reply:
But the process is only spawned if proxies have not been built yet - once proxies are compiled, we don't reach this section of code. Is this still a problem in this case (I don't know how the other packages work, so..)

Reply:
Ah OK, sorry, I only watched with one eye :)
The error message is much clearer, great improvement!

Reply:
OK, removing WIP. Final reviews please :)

Reply:
The two-lane way seems to work good. Just need to fix the strict version constraint.